### ISSUE TITLE
Label Leak during test time? Test Cloth ID == PID

### ISSUE THREAD

**Issue by ppriyank** [moon-wh/MADE#8]
Clothing labels are fed during test time and clothing labels during test time contain pid information 

PRCC : https://github.com/moon-wh/MADE/blob/master/data/datasets/prcc.py#L187C31-L187C60
LaST : https://github.com/moon-wh/MADE/blob/master/data/datasets/last.py#L152C17-L152C33

**Comment by moon-wh** [comment]
During testing, the pedestrian label has no direct relationship with the cloth ID.

**Comment by ppriyank** [comment]
Are you sure about this?
[PRCC LINE ](https://github.com/moon-wh/MADE/blob/master/data/datasets/prcc.py#L192)
```
clothes_id = pid2label[pid] * 2    #### THIS IS NOT ALLOWED !!
gallery_dataset.append((img_dir, pid, camid, clothes_id,imgdir2attribute[img_dir]))

clothes_id = pid2label[pid] * 2  #### THIS IS NOT ALLOWED !!
query_dataset_same_clothes.append((img_dir, pid, camid, clothes_id,imgdir2attribute[img_dir]))

clothes_id = pid2label[pid] * 2 + 1   #### THIS IS NOT ALLOWED !!
query_dataset_diff_clothes.append((img_dir, pid, camid, clothes_id,imgdir2attribute[img_dir]))
```

For LaST 
[LaST LINE](https://github.com/moon-wh/MADE/blob/master/data/datasets/last.py#L143C13-L153C90)
```
## relabel is False
clothes = names[0] + '_' + names[-1]
pid = int(names[0])
if relabel and pid2label is not None:
    pid = pid2label[pid]
if relabel and clothes2label is not None:
    clothes_id = clothes2label[clothes]
else:
    clothes_id = pid   #### THIS IS NOT ALLOWED !!
dataset.append((img_path, pid, camid, clothes_id,imgdir2attribute[img_path]))

```

PID == Cloth id during testing for these two datasets.

**Comment by xliangp** [comment]
Same issue here. During testing, clothing ID should not be used because it is unknown what the cloth the person is wearing. Yet testing used the same number of clothing ID as the training data set, which is strange because the testing data set has different clothes from the training dataset.

https://github.com/moon-wh/MADE/blob/facfb6e6acbfeeb188343849bb23fac81c591cbc/processor/processor.py#L243

**Comment by moon-wh** [comment]
Thanks for your attention. We referred to other papers and included cloth_id embeddings in our baseline model. Our method is independent of cloth_id. After discussions, we trained a model without cloth_id. In PRCC experiments, the performance difference between models with and without cloth_id embeddings is about 2%. I will update with a version excluding cloth_id embeddings and the corresponding results soon. You can also train after removing this part from the model. It is straightforward.

**Comment by moon-wh** [comment]
Thanks for your attention. We referred to other papers and included cloth_id embeddings as our baseline model. Our method is independent of cloth_id. After discussions, we trained a model without cloth_id. In PRCC experiments, the performance difference between models with and without cloth_id embeddings is about 2%. I will update with a version excluding cloth_id embeddings and the corresponding results soon. You can also train after removing this part from the model. It is straightforward.

**Comment by moon-wh** [comment]
Thanks for your attention. We referred to other papers and included cloth_id embeddings as our baseline model. Our method is independent of cloth_id. After discussions, we trained a model without cloth_id. In PRCC experiments, the performance difference between models with and without cloth_id embeddings is about 2%. I will update with a version excluding cloth_id embeddings and the corresponding results soon.

**Comment by ppriyank** [comment]
Why not have cloth id hardcoded as 0 during testing. Telling the model cloth ID makes “same cloth” evaluation obsolete

**Comment by ppriyank** [comment]
Do you happen to have any updates on this? None of the previous SOTA are using cloth IDs during testing. 
Can you report your official scores without using clothing ids?

**Comment by L-fountain** [comment]
Hello author, thank you for your work. We also encountered the same problem during the reproduction process, and when we hard coded everyone's cloth ID to 0, we obtained a result of R1 25%. May I ask if you can provide us with the modified test code you mentioned for our use.

**Comment by ppriyank** [comment]
Without clothing IDs (only EVA baseline) (removing clothing embedding or setting cloth id == 0)
In both cases, accuracy for PRCC is down by 4% on 60% Top-1 and 3-5% on LaST 77 Top-1.

Confirmed with other Person ReID paper authors as well, 
**During testing seeing clothing IDs is not allowed**

The same clothing evaluation looks for people with the same clothes and telling models the clothes labels eliminates IDS with different clothing ids, which is another label leak.    

**The current results for PRCC and LaST in paper are invalid**

**Comment by moon-wh** [comment]
Thanks for your suggestion. We have modified the data as described in the paper and updated the code accordingly. The new experimental results are all based on the baseline without the clothid encoding.
